### DISCUSSION TITLE
0.97.5.4 for MC 1.14.* & 1.15.* & 1.16.* & 1.17.* & 1.18.*

### DISCUSSION THREAD

**Discussion by LlmDl** [TownyAdvanced/Towny#5564] [Pre-Release Discussions]
- Added Custom TownBlockTypes, courtesy of Warriorrrr with PR #5388.
    - TownBlocks are now able to be created in the config and also via other plugins that use our API.
    - All of the previous townblocktypes will be imported to the new config section.
    - Each TownBlockType has a name, cost, tax, map character, itemuse list, switch list, and allowed blocks list.
  - New Command: /towny allowedblocks 
    - Used to display what blocks are permitted to be used on TownBlockTypes.
  - Removed Command: /towny farmblocks
    - Replaced with /towny allowedblocks.
  - Fix towns.ranks and nations.ranks missing from the Townyperms.yml's vital groups.
  - Fix for SQL schema and new townblock type-saving.
  - Fix support for spigot 1.18.1 enabling yaml comments by default.
    - Closes #5559.
  - Only apply the outsiders prevent pvp toggle when enabling pvp, courtesy of Warriorrrr with PR #5560.

----

<details><summary>Cumulative changes since 0.97.5.0</summary>
<details><summary>0.97.5.3</summary>

  - New Permission Node: towny.command.nation.default
    - A collection of nation sub commands which are given to everyone by default.
      - Each node doesn't do anything as there are further subcommands with more permission tests.
    - Fixes complicated townyperms setups not handling nation command refactor 0.97.3.2.
  - New Placeholder: %townyadvanced_player_location_plotgroup_name%
    - Shows the PlotGroup name if the player location has a PlotGroup.
    - Closes #5539.
  - Fix the default public town setting not being used.
  - Fix a typo in a lang key use, courtesy of Warriorrrr with PR #5540.
  - Allow economy.new_expand.price_claim_townblock_refund to be negative, meaning towns will be charged money to unclaim land.
    - Closes #3650.
  - Prevent the theft of axolotls, courtesy of Warriorrrr with PR #5542.
  - Fix minecart names in ItemLists, courtesy of Warriorrrr with PR #5546.
  - Fix ChorusFlowers not being protected from projectiles.
  - Fix transfering the last plot from a plotgroup into another leaving an empty plotgroup, vulnerable to errors.
  - New languages: es-MX, es-EC, es-CL, es-UY, es-VE.
  - API: Deprecated a bunch of abstract TownyDataSource methods that have been pointing to long-deprecated methods.
    - Older plugins might still be using these, instructions for updating are included in the javadocs.
  - Add in forwards compatibilty for server which update to the forthcoming custom townblocks, but then downgrade.
</details>
<details><summary>0.97.5.2</summary>

  - Fix language strings added to Towny via the API throwing errors for players using un-translated locales. 
  - New Command: /ta nation [nationname] transfer [townname]
    - Places the given town into the given nation, removing any old nation if needed.
    - Works toward closing #4906.
  - Fix bonemealing trees and grass not being aware of the player involved, and their permissions.
    - Closes #5517.
  - API: Add TownPreClaimEvent to initial TownBlock claimed during new Town code.
  - Fix lang\global.yml files generating empty on new installs.
  - Fix lang\override\ folder not being made on new installs.
  - Fix nations that change kings setting a null spawn when the new capital has a town spawn.
  - Replace underscores on override lang keys properly, courtesy of Warriorrrr with PR #5522.
    - Closes #5521.
  - Fix aborted teleports not refunding the spawn travel cost.
    - Closes #5526.
  - Add INK_SAC and GLOW_INK_SAC to ItemLists list of Dyes, to prevent them being used on Signs.
  - Automatic Config Update: 
    - RESPAWN_ANCHOR will be added to new_world_settings.plot_management.wild_revert_on_block_explosion.blocks,
    - Their explosions will be reverted in new worlds.
    - Your world files will also have it added, meaning existing worlds with block explosion revert set to on will start reverting Respawn Anchor explosions.
  - Make /ta set mayor TOWNNAME NPC be more careful about setting the NPC-run towns to have no upkeep.
  - Fix Title/Subtitles not showing when entering/exiting a town.
    - Closes #5532.
  - Translate Console givebonus warning, courtesy of GNosii with PR #5514.
  - Option to show what nation you're joining when you are invited to join a town, courtesy of AlessioGr with PR #5530. (First-Time Contributor!)
</details>
<details><summary>0.97.5.1</summary>

  - Add helpful methods to BaseCommand.
  - Fix regex used to split the townyperms.yml error message.
  - API: Added StatusScreenEvent#getCommandSender()
    - It is now possible to learn who is going to view the StatusScreen being made.
  - Fix expired invite handler not accounting for sent nation ally invites.
  - New /ta merging commands, courtesy of Gamer153 with PR #5491. (First-Time Contributor!)
  - New Command: /ta town [townname] merge [townname] - Asks the 2nd town if they want to merge with the first.
  - New Command: /ta town [townname] forcemerge [townname] - Forcefully merges the 2nd town into the first.
  - New Command: /ta nation [nationname] merge [nationname] - Asks the 2nd nation if they want to merge with the first.
  - New Command: /ta nation [nationname] forcemerge [nationname] - Forcefully merges the 2nd nation into the first.
  - Make town neutrality cost appear on /towny prices for #5500.
  - Add town and nation neutrality costs to their respective status screens.
    - Closes #5500.
  - Reorganization and cleanup of Town, Nation, and TownySettings, courtesy of FlagCourier with PR #5493.
  - Made it possible for the {townname} placeholder to be used in the wilderness_title and wilderness_subtitle messages.
  - Townyperms.yml nations' default rank will automatically be given towny.command.nation.spawn.
  - Add 2 convenience methods to PreNewTownEvent, courtesy of Warriorrrr with PR #5504.
  - Fix possible tab completion NPE, courtesy of Warriorrrr with PR #5506.
  - Allow lava_cauldrons to kill minecarts.
    - Closes #5503.
  - Call the PreNewTownEvent after any confirmations are confirmed, courtesy of Warriorrrr with PR #5508.
  - API: Add tab-completion helper methods to TownyAPI.
</details>
</details>

----

<table align=center>
<tr><th colspan=5><h3>Towny Sponsors<br><h4>I want to give a big thank you to all of my sponsors:<br>These are the people that help to make Towny's support and development as active as it is,<br>and who supported me during this pre-release of Towny. </th></tr>
<tr><td>MrLogano</td><td>Momshroom</td><td>Cobrex1</td><td> KarlOfDuty </td><td>scatterspell</td></tr>
<tr><td>TheFlagCourier</td><td>gerardvanschip</td><td>DrprofLuigi</td><td>Tobza</td><td>SmallSansSerif</td></tr>
<tr><td>TheNylox</td><td>diontimmer</td><td>trevor1097</td><td>benevolent2012</td><td>bretomart</td></tr>
<tr><td>0xBit-MC</td><td>specs1396</td><td>treeDerevo</td><td>Scottlander100</td><td>ItsSynchro</td></tr>
<tr><td>EnvyEnfys</td><td>MidnightSugar</td><td>mzyax</td><td>PestoNotPasta</td><td>cigaming</td></tr>
<tr><td>KeNickGong</td><td>honnisha</td><td>MilesBHuff</td><td>pebblehost</td><td>Akiranya</td></tr>
<tr><td>BamBamTheEggman</td><td>NoikzGaming</td><td>TakeoRenka</td><td>LordHenryVonHenry</td><td>nUSAHudson</td></tr>
<tr><td>SolarCiv</td><td>Litning11</td><td>Black-Tongue</td><td>SkapMC</td><td>Tauses</td></tr>
<tr><td>AaronBDesign</td><td>LeftTheNegotiator</td><td>MaluuhLive</td><td>anaxios</td><td>Zectogonix</td></tr>
<tr><td>Lemuridaee</td><td>ReEvadere</td><td>bedrye</td><td>qckcat</td><td>roflmarv</td></tr>
<tr><td>Tomboru</td><td>beabuenodev</td><td>PawsTenebria</td><td>BirbEXE</td><td>LoJoSho</td></tr>
<tr><td>79372275372</td><td>benjamin1313</td><td>ThePontificate</td><td>AllieDragon</td><td>TegulaThePig</td></tr>
<tr><td>SavvaKulikov</td><td>brycenaddison</td><td>benlmyers</td><td colspan=2>and 11 private sponsors.</td></tr>
<tr><td colspan=5 align=center>
<h3><a href=https://github.com/sponsors/LlmDl>If you want to support the developer, become a sponsor.</a></h3>
(It's just like Patreon but instead 100% of your support goes to the developer.)
<h3>Plus there are <a href="https://github.com/TownyAdvanced/Towny/wiki/Sponsor-Perks">Sponsor Perks!</a></h3>
</td></tr>
</table>

<table align=center>
<th colspan=3> Important Links </th>
<tr align=center><td><a title="How Towny Works" href="https://github.com/TownyAdvanced/Towny/wiki/How-Towny-Works">How Towny 
Works</a></td><td><a title="Install Guide" href="https://github.com/TownyAdvanced/Towny/wiki/Installation">Towny Install Guide</a></td><td><a title="Updating Towny" href="https://github.com/TownyAdvanced/Towny/wiki/Updating-Towny">Towny Update Guide</a></td>
</tr>
<tr align=center>
<td><a title="Towny Changelog" href="https://raw.githubusercontent.com/TownyAdvanced/Towny/master/resources/ChangeLog.txt" target="_blank" rel="noopener">Complete Changelog</a> </td><td><a title="Default Config Files" href="https://github.com/TownyAdvanced/Towny/wiki/Config-Files">Default Config Files</a></td><td><a title="Commands/Permissions/Placeholders" href="https://github.com/TownyAdvanced/Towny/wiki/Reference">Commands/Permissions<br>Placeholders</a></td>
</tr>
</table>

### ⏬ Download available as a .jar file in the Assets section below:

<hr /><em>This discussion was created from the release <a href='https://github.com/TownyAdvanced/Towny/releases/tag/0.97.5.4'>0.97.5.4 for MC 1.14.* & 1.15.* & 1.16.* & 1.17.* & 1.18.*</a>.</em>

**Comment by benlmyers** [comment]
Thank you for the quick fix!
